integration-tests/stringtemplate/src/test/java/org/apache/camel/quarkus/component/stringtemplate/it/StringtemplateIT.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.component.stringtemplate.it;
+
+import java.util.Collections;
+
+import io.quarkus.test.junit.NativeImageTest;
+import io.restassured.RestAssured;
+import io.restassured.http.ContentType;
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.CoreMatchers.containsString;
+
+@NativeImageTest
+class StringtemplateIT extends StringtemplateTest {
+
+    @Test
+    public void testDisabledContextMapAll() {
+        //allowContextMapAll has to be disabled in native mode
+        RestAssured.given()
```

Comment:
I've added test for native to verify that parameter `allowContextMapAll` is disabled. 
But because there are usually no special tests for native mode, I'd like to ask. @jamesnetherton , @ppalaga   Should I keep it or remove it?

Reply:
No problem with having a special test for native. We definitely had such in the past, not sure we still have them elsewhere.

Reply:
We should agree on whether [this](https://github.com/apache/camel-quarkus/blob/bd845d3959984d575441d3e79cd9a58095b6aa34/extensions/stringtemplate/runtime/src/main/java/org/apache/camel/quarkus/component/stringtemplate/StringtemplateEndpointSubstitute.java) is the right thing to do.

Originally, I saw the 'fix' to #1807 as purely a documentation exercise. Where we just state for each affected extension that `contextMapAll` will not work in native mode.

Potentially forcing `contextMapAll` to `false` is a bit confusing because you would run into problems at runtime. Where as without it, the native build would fail, so it's sort like a 'fail fast'.

Reply:
So, removing substitution (which forcescontextMapAll = true), rewriting limitations (that exchange.properties won't work) and removing test, should be the correct solution, yes?

Reply:
IMO, yes. @ppalaga WDYT?

Reply:
I think about documentation like this one:
> If property `allowContextMapAll` is set to `true` there is a limitation in native mode. Reflective calls to methods
of several classes in camel are not allowed (for example `org.apache.camel.support.DefaultExchange`). Therefore 
some expression won't work (for example `<exchange.properties.*>).
>
>You can allow reflective calls to such classes (for example by application.properties), but keep in mind that it brings 
some security risks.

WDYT? @ppalaga , @jamesnetherton

Reply:
+1 for removing the substitution and letting the native compilation fail

Reply:
Sounds ok to me. Maybe tweak `some expression` to `some expressions`.

Reply:
done